pkg/api/v1alpha1/iperf3_types.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2019 The xridge kubestone contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Iperf3Spec defines the Iperf3 Benchmark Job
+type Iperf3Spec struct {
+	// time in seconds to transmit for
+	Time int32 `json:"time,omitempty"`
+
+	// Use UDP rather than TCP
+	// +optional
+	UDP bool `json:"udp,omitempty"`
+}
+
+// Iperf3Status defines the observed state of Iperf3
+type Iperf3Status struct {
```

Comment:
The similar statuses in kubernetes are handled by enums (e.g. pod status). I think a single status with the possible values of "Running", "Completed", "Failed", etc. would be more idiomatic.

Reply:
go does not have enums unfortunately. the best we can do to replicate is to create string constants (which is eeh).
You also can find this pattern in Kubernetes, take a look at job status:
```
apiVersion: batch/v1
kind: Job
...
status:
  active: 1
  startTime: "2019-07-31T07:35:44Z"
```
and:
```
apiVersion: batch/v1
kind: Job
metadata:
  creationTimestamp: "2019-07-29T10:00:09Z"
...
status:
  completionTime: "2019-07-29T20:00:08Z"
  conditions:
  - lastProbeTime: "2019-07-29T20:00:08Z"
    lastTransitionTime: "2019-07-29T20:00:08Z"
    status: "True"
    type: Complete
  startTime: "2019-07-29T10:00:09Z"
  succeeded: 1
```